torch/distributed/tensor/parallel/ddp.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, List, Tuple
+
+import torch.nn as nn
+from torch.distributed.tensor.parallel.fsdp import _flatten_tensor, _unflatten_tensor
+
+__all__ = ["pre_dp_model_transform"]
+
+
+def _get_submodule_n_params(module: nn.Module, path: str):
+    """
+    Get submodule and the direct path of parameter from the module
+    """
+    if "." in path:
+        path_list = path.split(".")
+        parent_module_path = ".".join(path_list[:-1])
+        module = module.get_submodule(parent_module_path)
+        path = path_list[-1]
+    return module, path
+
+
+def _update_model_param(param_list: List[Tuple[nn.Module, str, nn.Parameter]]):
+    """
+    Update parameters within the model
+    """
+    for item in param_list:
+        parent_module, module_path, t = item
+        assert hasattr(parent_module, module_path)
+        delattr(parent_module, module_path)
+        setattr(parent_module, module_path, t)
+
+
+def _reconstruct_dtensor(model: nn.Module, _input: Any):
+    """
+    Recontruct DTensor parameters from local tensors
+    """
+    param_list = []
+    for name, t in model.named_parameters():
```

Comment:
If we have DDP above FSDP, then will this try to convert all FSDP parameters to DTensor as well?

Reply:
No, we only convert parameters which are DTensors.

Reply:
What I meant is that, would this convert DTensors under an FSDP-managed module, not a DDP-managed module?

Reply:
For now, no. We hope down the road, we can merge FSDP-managed module into this API, too.

Reply:
My point is that `model.named_parameters()` recurses into submodules, which maybe managed by FSDP. There is no check against that to stop the recursion. This means that if there is a DDP module above FSDP modules, then this will convert the FSDP-managed DTensors to local tensor too. Is that the desired behavior?